### PR TITLE
[Test] gateway e2e responder handshake testing (no noise)

### PR DESCRIPTION
**Pull request by niklaslong**
A replacement PR for #2660 without the noise changes. Increases gateway handshake test coverage to various cases including:
- timeouts
- invalid challenge request/response
- unexpected disconnects

The happy path is exercised in all our e2e tests already so this changeset focusses on the failure cases. 

Drafted for now (I may add a few more cases) but comments already welcome!

### CODE REVIEW

**Comment by niklaslong** [comment]
Note, we may need to increase the timeouts slightly if the tests end up being flaky on CI.

**Comment by niklaslong** [comment]
Rebased. Note the bft integration tests will fail as they currently do on `mainnet-staging`.

**Comment by niklaslong** [comment]
Rebased on top of #3174 to fix the build, the last 6 commits are new.

File: node/tests/common/test_peer.rs [hunk 1]
```diff
@@ -113,7 +113,6 @@ impl TestPeer {
     }
 }
 
-#[async_trait::async_trait]
```
Comment (joske) [inline node/tests/common/test_peer.rs:116]: didn't we say to keep this for now?

Comment (niklaslong) [reply node/tests/common/test_peer.rs:116]: Yeah, I haven't touched it in the rest of the codebase but I did update pea2pea which no longer uses these traits.

**Review by howardwu — APPROVED** [review]

**Pull request merged**